airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultReplicationWorker.java

```diff
@@ -215,8 +220,8 @@ private void replicate(final Path jobRoot,
             }
           });
 
-      final CompletableFuture<?> readSrcAndWriteDstThread = CompletableFuture.runAsync(
-          readFromSrcAndWriteToDstRunnable(
+      final CompletableFuture<?> readSrcAndWriteDstThread = srcHeartbeatTimeoutChaperone
```

Comment:
@gosusnp I was needing a CompletableFuture here because of the log on line 250 and 252.